utils/wasm-proc/src/main.rs

```diff
@@ -57,10 +55,61 @@ fn main() {
             "meta_init_output",
             "meta_title",
         ],
-    )
-    .expect("Metadata optimizer failed");
-    parity_wasm::serialize_to_file(metadata_file_name.clone(), metadata_module)
-        .expect("Serialization failed");
+    ) {
+        println!("Optimizer failed");
+    }
+
+    if let Err(e) = parity_wasm::serialize_to_file(metadata_file_name.clone(), metadata_module) {
+        println!("Serialization failed: {}", e);
+    }
 
     println!("Metadata wasm: {}", metadata_file_name.to_string_lossy());
 }
+
+fn main() {
+    let meta = Arg::with_name("meta")
+        .short("m")
+        .long("meta")
+        .takes_value(false)
+        .help("Provides a metadata .meta.wasm file");
+
+    let opt = Arg::with_name("optimize")
+        .short("o")
+        .long("optimize")
+        .takes_value(false)
+        .help("Provides an optimized .opt.wasm file");
+
+    let path = Arg::with_name("path")
+        .short("p")
+        .long("path")
+        .required(true)
+        .index(1)
+        .takes_value(true)
+        .multiple(true)
+        .help("Specifies path to .wasm file(-s)");
+
+    let app = App::new("wasm-proc").args(&[meta, opt, path]);
+
+    let matches = app.get_matches();
+
+    let wasm_files: Vec<&str> = matches
+        .values_of("path")
+        .expect("Path to wasm files is required")
+        .collect();
+
+    let o = matches.is_present("optimize");
+    let m = matches.is_present("meta");
+
+    for file in wasm_files {
+        if let Ok(module) = parity_wasm::deserialize_file(file) {
+            if o || !(o || m) {
+                optimize(file, module.clone());
+            }
+            if m || !(o || m) {
+                optimize_meta(file, module.clone());
+            }
+        } else {
+            println!("Failed to load wasm file: {}", file);
```

Comment:
make `main()` return Result<(), ?>

if you print errors this way, exit code of the program would be 0

Reply:
Yes, nothing crushes, we have an error in console, and script continues work with other arguments. Do you want to break processing of the script if we ran into error?

Reply:
Yes, like every sensible program does

Reply:
Done